src/pixie/fileformats/qoi.nim

```diff
@@ -0,0 +1,193 @@
+import std/endians, chroma, flatty/binny
+import pixie/[common, images, internal]
+
+# See: https://qoiformat.org/qoi-specification.pdf
+
+const
+  qoiSignature* = "qoif"
+  indexLen = 64
+  opRgb = 0b11111110'u8
+  opRgba = 0b11111111'u8
+  opMask2 = 0b11000000'u8
+  opIndex = 0b00000000'u8
+  opDiff = 0b01000000'u8
+  opLuma = 0b10000000'u8
+  opRun = 0b11000000'u8
+
+type
+  Header = tuple
+    width, height: uint32
+    channels, colorspace: uint8
+  Colorspace* = enum sRBG = 0, linear = 1
+  Index = array[indexLen, ColorRGBX]
+
+func hash(p: ColorRGBX): int =
+  (p.r.int * 3 + p.g.int * 5 + p.b.int * 7 + p.a.int * 11) mod indexLen
+
+proc decodeQoiRaw*(data: string): (Image, Colorspace) {.raises: [PixieError].} =
```

Comment:
Please do not use an `Image` to hold straight alpha data. It does not ever hold that. If you want to hold straight alpha values, please make a type for that similar to `Png` in the png.nim file.

Reply:
Good idea.